File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def changeNum(n, k):
    num = []
    
    while k <= n:
        num.append(str(n % k))
        n //= k
    num.append(str(n % k))

    num.reverse()
    
    return num

def isPrime(n):
    for i in range(2, int(n**(1/2)) + 1):
        if n % i == 0:
            return False
    return True
        
def solution(n, k):    
    numStr = ''.join(changeNum(n,k))
    numList = numStr.split('0')
    print(numList)
    
    answer = []
    for num in numList:
        if num == '' or num =='1':
            continue
        if isPrime(int(num)):
            answer.append(num)
    
    return len(answer)

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function solution(n, k) {
    return get_k_base(n, k)
        .split('0')
        .filter((base) => get_prime(base))
        .length;;
}

function get_k_base(n, k) {
    const base = [];
    
    while(n) {
        base.push(n % k);
        n = Math.floor(n / k);
    }
    
    return base.reverse().join('');
}

function get_prime(n) {
    if (n < 2) return false;
    
    for (let i = 2; i <= Math.sqrt(n); i++) {
        if (n % i === 0) return false;
    }
    
    return true;
}

